src/timer/mod.rs

```diff
@@ -1,29 +1,28 @@
+use super::callback::Callback;
 use super::linked_list::{LinkedList, LinkedListNode};
 use core::cell::Cell;
 
 #[cfg(test)]
 mod test;
 
-pub struct TimerData {
+pub struct TimerData<'a> {
     remaining_ticks: Cell<u32>,
-    context: Cell<Option<*const ()>>,
-    callback: Cell<Option<fn(*const ())>>,
+    callback: Cell<Option<Callback<'a>>>,
 }
 
-impl TimerData {
+impl TimerData<'_> {
```

Comment:
What does this do for us?

Reply:
pretty sure we need it because the callback now tracks that the context is alive while it is alive, before you could hide that with the way it was used, as for why I used `'_` simply because rustc is able to guess that one, but it does not allow an implicit lifetime

Reply:
Got it, I didn't realize this was necessary, I thought you may have added it to solve a problem I didn't understand.